src/note_bariolage.lua

```diff
@@ -0,0 +1,88 @@
+
+function plugindef()
+  -- This function and the 'finaleplugin' namespace
+  -- are both reserved for the plug-in definition.
+  finaleplugin.Author = "Jacob Winkler"
+  finaleplugin.Copyright = "2022"
+  finaleplugin.Version = "1.0"
+  finaleplugin.Date = "3/18/2022"
+  finaleplugin.Notes = [[
+  USING THE 'BARIOLAGE' SCRIPT
+  
+This script creates bariolage-style notation where layers 1 and 2 interlock. It works well for material that has even-numbered beam groups like 4x 16th notes or 6x 16th notes (in compound meters). 32nd notes also work. Odd numbers of notes produce undesirable results.
+
+To use, create a suitable musical passage in layer 1, then run the script. The script does the following:
+- Duplicates layer 1 to layer 2.
+- Mutes playback of layer 2.
+- Iterates through the notes in layer 1. For even-numbered notes (i.e. the 2nd and 4th 16ths in a group of 4) it replaces the stem with a blank shape, effectively hiding it.
+- Any note in layer 1 that is the last note of a beamed group is hidden.
+- Iterates through the notes in layer 2 and changes the stems of the odd-numbered notes.
+- Any note in layer 2 that is the beginning of a beamed group is hidden.
+]]
+  return "Bariolage", "Bariolage", "Bariolage: Creates alternating layer pattern from layer 1. Doesn't play nicely with odd numbered groups!"
+end
+
+function layer_copy(src, dest)
+    local region = finenv.Region()
+    local start=region.StartMeasure
+    local stop=region.EndMeasure
+    local sysstaves = finale.FCSystemStaves()
+    sysstaves:LoadAllForRegion(region)
+    src = src - 1
+    dest = dest - 1
+    for sysstaff in each(sysstaves) do
+        staffNum = sysstaff.Staff
+        local noteentrylayerSrc = finale.FCNoteEntryLayer(src,staffNum,start,stop)
+        noteentrylayerSrc:Load()     
+        local noteentrylayerDest = noteentrylayerSrc:CreateCloneEntries(dest,staffNum,start)
+        noteentrylayerDest:Save()
+        noteentrylayerDest:CloneTuplets(noteentrylayerSrc)
+        noteentrylayerDest:Save()
+    end
+end -- function layer_copy
+
+function stems_hide(entry)
```

Comment:
Same thing here. This would make a great library function `note_entry.hide_stem`

Reply:
How about note_entry.stem_hide? Feels like that's more in line with the philosophy of listing 1)The Thing and then 2)What To Do With It...

Reply:
thing.what()

Reply:
never mind, I agree hide_stem is better after looking more closely...